### PR TITLE
ignore hidden values when conducting association tests

### PR DESCRIPTION
**Pull request by congyu-lu**
## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: client/termdb/TermdbVocab.js [hunk 1]
```diff
@@ -172,6 +172,17 @@ export class TermdbVocab extends Vocab {
 		if ('grade' in opts) params.push(`grade=${opts.grade}`)
 		if ('minSampleSize' in opts) params.push(`minSampleSize=${opts.minSampleSize}`)
 
+		if (opts.term2) {
```
Comment (gavrielm) [inline client/termdb/TermdbVocab.js:175]: Would be useful to have a short comment here explaining why this is done

File: server/src/termdb.barchart.js [hunk 2]
```diff
@@ -634,9 +634,20 @@ Output: The function has no return but appends the statistical results to the pr
 	]
 }
 */
-async function computePvalues(data, ds) {
+async function computePvalues(data, ds, hiddenValues) {
+	// remove hidden values from deep copy of data
+	const charts = JSON.parse(JSON.stringify(data.charts))
+	for (const chart of charts) {
+		chart.serieses = chart.serieses.filter(s => !hiddenValues.term1.includes(s.seriesId))
```
Comment (gavrielm) [inline server/src/termdb.barchart.js:641]: I don't think you should replace the same array with filtered array. Would be better to create new array.

File: server/src/termdb.barchart.js [hunk 3]
```diff
@@ -634,9 +634,20 @@ Output: The function has no return but appends the statistical results to the pr
...
+		for (const s of chart.serieses) {
+			s.data = s.data.filter(d => !hiddenValues.term2.includes(d.dataId))
```
Comment (gavrielm) [inline server/src/termdb.barchart.js:643]: Same issue as above

File: server/src/termdb.barchart.js [hunk 4]
```diff
@@ -634,9 +634,20 @@ Output: The function has no return but appends the statistical results to the pr
 	]
 }
 */
-async function computePvalues(data, ds) {
+async function computePvalues(data, ds, hiddenValues) {
+	// remove hidden values from deep copy of data
+	const charts = JSON.parse(JSON.stringify(data.charts))
```
Comment (gavrielm) [inline server/src/termdb.barchart.js:639]: You should just create a new object from scratch called `charts`. You can then iterate through `data.charts` and populate `charts` with the relevant information.

Comment (congyu-lu) [reply server/src/termdb.barchart.js:639]: Thanks, I will do that

**Review by gavrielm — APPROVED** [review]
Works well. Thank you for implementing this!

**Pull request merged**